src/marten/db/field/slug.cr

```diff
@@ -29,8 +51,18 @@ module Marten
           end
         end
 
-        macro check_definition(field_id, kwargs)
-          # No-op max_size automatic checks...
+        private def generate_slug(value)
+          suffix = "-#{Random::Secure.hex(4)}"
+
+          slug = value.gsub(NON_ALPHANUMERIC_RE, "").downcase
+          slug = slug.gsub(WHITESPACE_HYPHEN_RE, "-").strip("-_")
+          slug = slug.gsub(/[^\x00-\x7F]/, "")
+
+          slug[...(max_size - suffix.size)] + suffix
+        end
```

Comment:
It would be interesting to extract any slug generation logic in a dedicated concern module (eg. `Marten::Core::Sluggable`). This would allow it to be easily reused in other parts of the framework but also by users of the framework. WDYT?